packages/edit-post/src/plugins/copy-content-menu-item/index.js

```diff
@@ -8,27 +8,29 @@ import { withState, compose } from '@wordpress/compose';
 
 function CopyContentMenuItem( { createNotice, editedPostContent, hasCopied, setState } ) {
 	return (
-		<ClipboardButton
-			text={ editedPostContent }
-			role="menuitem"
-			className="components-menu-item__button"
-			onCopy={ () => {
-				setState( { hasCopied: true } );
-				createNotice(
-					'info',
-					'All content copied.',
-					{
-						isDismissible: true,
-						type: 'snackbar',
-					}
-				);
-			} }
-			onFinishCopy={ () => setState( { hasCopied: false } ) }
-		>
-			{ hasCopied ?
-				__( 'Copied!' ) :
-				__( 'Copy All Content' ) }
-		</ClipboardButton>
+		editedPostContent.length > 0 && (
```

Comment:
`getEditedPostAttribute` is a non-trivial selector, but it was previously more necessary because we did not conditionally render the element based on whether there is content or not.

With the addition of this condition, I wonder if we might consider to optimize the rendering to check whether there is content using the more-optimized [`isEditedPostEmpty`](https://github.com/WordPress/gutenberg/blob/b95e865de32ee62d722be3e3e71b334a4b639032/packages/editor/src/store/selectors.js#L474-L512) (see inline comments for optimization specifics).

It could be applied using a series of higher-order components:

```js
export default compose(
	withSelect( ( select ) => ( {
		hasEditedPostContent: ! select( 'core/editor' ).isEditedPostEmpty(),
	} ) ),
	ifCondition( ( { hasEditedPostContent } ) => hasEditedPostContent ),
	withSelect( ( select ) => ( {
		editedPostContent: select( 'core/editor' ).getEditedPostAttribute( 'content' ),
	} ) ),
	withDispatch( ( dispatch ) => {
		const {
			createNotice,
		} = dispatch( 'core/notices' );

		return {
			createNotice,
		};
	} ),
	withState( { hasCopied: false } )
)( CopyContentMenuItem );
```